src/pydvl/utils/numeric.py

```diff
@@ -59,21 +60,29 @@ def num_samples_permutation_hoeffding(eps: float, delta: float, u_range: float)
     return int(np.ceil(np.log(2 / delta) * 2 * u_range**2 / eps**2))
 
 
-def random_subset(s: NDArray[T], q: float = 0.5) -> NDArray[T]:
+def random_subset(
+    s: NDArray[T],
+    q: float = 0.5,
+    seed: numbers.Integral | np.random.SeedSequence | np.random.Generator = None,
```

Comment:
> This syntax is only supported starting from 3.10. You should use `Union` instead

Note that we use `from future import __annotations__` in several files to support this operator.

Reply:
I see. I think that for the sake of consistency we should continue the types from the typing module and leave this change for another issue.

Reply:
> I see. I think that for the sake of consistency we should continue the types from the typing module and leave this change for another issue.

Since there are several files using it, it's already too late. We can systematically add it at some point, but I don't think it's a big issue.

Reply:
Already reverted it.